Split --uktdd-body-position-try-fallbacks into two fixed slots (#404)
acusti/uikit: packages/dropdown/README.md

### Changing the Default Direction

`--uktdd-body-position-area` and its two fallback slots
(`--uktdd-body-position-try-fallback-1`,
`--uktdd-body-position-try-fallback-2`) always change together.
`--uktdd-body-position-area` is the primary placement, used whenever it
fits in the viewport — that’s what actually determines the direction a
dropdown opens by default. The fallback slots are only consulted when the
primary placement doesn’t fit, so overriding them alone changes nothing in
the common case where the primary already fits. Overriding the primary
alone works, but leaves behind fallback slots tuned for the old primary —
in a cramped viewport, the dropdown can fall back toward the direction you
just moved away from — so set all three together, and update both slots. If
the new direction has only one meaningful fallback (a centered menu, say),
put it in slot 1 and set slot 2 to the shipped no-op `--uktdd-noop` so it

> doesn’t inherit that flip.

Author: ```suggestion
dropdown opens by default. The fallback slots are tried in order only when the
primary placement doesn’t fit. Set all three together to ensure the menu behaves
the way you want to in all possible contexts. If the new direction has only one
meaningful fallback (a centered menu, say), put it in slot 1 and set slot 2 to the
shipped no-op `--uktdd-noop` so it doesn’t inherit that flip.
```

Author: Applied in 38351e75.

### Changing the Default Direction

`--uktdd-body-position-area` and its two fallback slots
(`--uktdd-body-position-try-fallback-1`,
`--uktdd-body-position-try-fallback-2`) always change together.
`--uktdd-body-position-area` is the primary placement, used whenever it
fits in the viewport — that’s what actually determines the direction a
dropdown opens by default. The fallback slots are only consulted when the
primary placement doesn’t fit, so overriding them alone changes nothing in
the common case where the primary already fits. Overriding the primary
alone works, but leaves behind fallback slots tuned for the old primary —
in a cramped viewport, the dropdown can fall back toward the direction you
just moved away from — so set all three together, and update both slots:
leaving one at its default inherits a flip tuned for the old primary, the
same footgun on a smaller scale. If the new direction has only one
meaningful fallback (a centered menu, say), put it in slot 1 and set slot 2

> to the shipped no-op `--uktdd-noop` so it doesn’t inherit that flip.

Author: how about:
```suggestion
just moved away from — so set all three together, and update both slots.
If the new direction has only one meaningful fallback (a centered menu,
say), put it in slot 1 and set slot 2 to the shipped no-op `--uktdd-noop`
so it doesn’t inherit that flip.
```

Author: Applied your suggestion in e5a75846.

`block-start`/`block-end` read as `top`/`bottom` in the near-universal
horizontal-tb writing mode.

Pick the row matching the direction you want, and set the three variables

> to its values:

Author: ```suggestion
Pick the row matching the direction you want, and set the three CSS custom
properties to the corresponding values:
```

Author: Applied in 38351e75.

roomier side and scrolls rather than overflowing.

There are exactly two author fallback slots by design. The spec lets
engines cap the position options list at an implementation-defined length
with a floor of five — and that list includes the element’s base position,
so only four fallbacks past it are guaranteed. The component appends the
two fill placements after your two slots, which is exactly four — within
the floor on every conformant engine. That’s why the fallbacks are two
separate custom properties rather than a list: to avoid a third (or more)
entry, which could result in a trigger near a viewport edge opening
off-screen instead of flipping and scrolling. The two default slots are the
single-axis flips of the primary (a block flip and an inline flip); the
both-axes diagonal is intentionally not a slot, so a trigger crammed into
the corner diagonal to the primary opening direction lands in a fill rather
than a natural placement — re-tune the primary (per the recipes above) for
triggers pinned to a viewport corner. For a recipe with a single meaningful
fallback (a centered menu, say), set slot 1 and leave slot 2 as the shipped

> no-op `--uktdd-noop`, so it doesn’t inherit the default’s flip.

Author: ```suggestion
There are exactly two author fallback slots by design. The [CSS spec](https://drafts.csswg.org/css-anchor-position-1/#fallback-apply)
requires implementations to support at least five position options,
including the element’s base position, and `@acusti/dropdown` appends
two last-resort placements to the end, so that leaves two fallbacks that
you can specify, hence why the are exposed as two separate custom
properties. The two default slots are the
single-axis flips of the primary (a block flip and an inline flip); the
mirror opposite diagonal is intentionally not a slot, so that a trigger crammed into
the corner diagonal to the primary opening direction lands in a fill rather
than a natural placement. See the [recipes above](#changing-the-default-direction)
for different common placement scenarios. For a recipe with a single meaningful
fallback (a centered menu, say), set slot 1 and leave slot 2 as the shipped
no-op `--uktdd-noop`, so it doesn’t inherit the default’s flip.
```

Author: Applied in 38351e75 (fixed one typo in the suggestion: 'hence why the are exposed' → 'they are exposed').

full by the named placements) can’t reject both fills: the body fills the
roomier side and scrolls rather than overflowing.

There are exactly two author fallback slots by design. The spec lets
engines cap the position options list at an implementation-defined length
with a floor of five — and that list includes the element’s base position,
so only four fallbacks past it are guaranteed. The component appends the
two fill placements after your two slots, which is exactly four — within
the floor on every conformant engine. That’s why the fallbacks are two
separate custom properties rather than a list: a list invites a third
entry, which would push a fill past the floor and let a trigger near a

> viewport edge open off-screen instead of flipping and scrolling. The two

Author: maybe (will require re-formatting the file):
```suggestion
separate custom properties rather than a list: to avoid a third (or more)
entry, which could result in a trigger near a viewport edge opening
off-screen instead of flipping and scrolling. The two
```

Author: Applied your suggestion in e5a75846 (reformatted).

```css
.tall-menu {
    --uktdd-body-position-try-fallback-1: --uktdd-top-start;
    --uktdd-body-position-try-fallback-2: --uktdd-bottom-end;
    --uktdd-body-fill-fallbacks:

>         --uktdd-fill-cover, --uktdd-fill-bottom, --uktdd-fill-top;

Author: There's no contradiction here — the "five" in the PR/changesets is the **spec floor** (five *options* = the base position + four fallbacks), not Chromium's limit. Chromium evaluates exactly **six** options (base + five fallbacks); this was measured directly with a size-based-rejection test harness, and Safari 26.5 and Firefox 152 both evaluate well past twelve. The fill-fallbacks changeset states it explicitly: "evaluating six options (five fallbacks past the base)."

So the guidance is accurate and already balanced: the default list stays within the spec-guaranteed five-option floor, and the opt-in cover recipe (six options) is one above that floor — safe on every current engine, with the section immediately noting you can drop to one author fallback to keep the whole list within the spec floor. Removing the empirical per-engine data would undo a deliberate de-hedge made after that cross-engine testing.

properties follow the established naming scheme:

- `--uktdd-submenu-position-area` (default: `inline-end span-block-end`)
- `--uktdd-submenu-position-try-fallback` (default:
  `--uktdd-submenu-inline-start`; a single slot, not a list — the component
  appends two last-resort fills after it, and the spec guarantees only four

>   evaluated fallbacks, so there’s only room for the one author fallback)

Author: let’s just say
```suggestion
- `--uktdd-submenu-position-try-fallback` (default:
  `--uktdd-submenu-inline-start`; a single slot, not a list)
```

Author: Applied your suggestion in e5a75846.

  out of scope for a macOS-style menu’s narrow columns — is the one case
  this gives up versus covering the parent; append
  `--uktdd-fill-bottom, --uktdd-fill-top` here to restore the
  parent-covering rescue if you need it, leaving
  `--uktdd-submenu-position-try-fallback` at its default single option to

>   stay within the evaluation limit)

Author: how about:
```suggestion
default. A submenu wider than the space on either side of its parent is
the one case this default doesn’t handle; append
`--uktdd-fill-bottom, --uktdd-fill-top` here to restore the
  parent-covering rescue if you need it
```

Author: Applied your suggestion in e5a75846.
